akka-remote/src/main/scala/akka/remote/artery/ImmutableLongMap.scala

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import scala.annotation.tailrec
+import akka.util.OptionVal
+import scala.reflect.ClassTag
+import java.util.Arrays
+
+/**
+ * INTERNAL API
+ */
+private[akka] object ImmutableLongMap {
+  def empty[A >: Null](implicit t: ClassTag[A]): ImmutableLongMap[A] =
+    new ImmutableLongMap(Array.emptyLongArray, Array.empty)
```

Comment:
@2beaucoup I have still not changed this, because I feel uncertain about that `ClassTag`.
This seems to work:
```
  private val _empty = new ImmutableLongMap(Array.emptyLongArray, Array.empty)

  def empty[A >: Null](implicit t: ClassTag[A]): ImmutableLongMap[A] =
    _empty.asInstanceOf[ImmutableLongMap[A]]
```

but it feels wrong with regards to the passed in `ClassTag`